drivers/serial/uart_nrfx_uart.c

```diff
@@ -395,6 +454,14 @@ static int uart_nrfx_init(struct device *dev)
 
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
 
+	/* The aforementioned API functions will always return true before
+	 * the driver has transmitted any characters, regardless of the status
+	 * of the TX_READY event flag. After that the functions will always
+	 * return NRF_UART_EVENT_RXDRDY after the driver has transmitted
```

Comment:
This comment should be rephrased. I don't understand which "aforementioned" functions you refer to, what is the "TX_READY event flag", and what `NRF_UART_EVENT_RXDRDY` has to do with all of this.